packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/MutableRealmTests.kt

```diff
@@ -675,4 +675,28 @@ class MutableRealmTests {
             }
         }
     }
+
```

Comment:
You can try to do this instead:

```
fun deleteAll_shouldThrowOnClosedRealm() { 
  realm.write { 
    realm.close()  
    assertFailsWith<IllegalStateException> { deleteAll() } 
  } 
}
```

But I'm not sure that is a use case we care about, e.g. `MutableRealm.delete()` doesn't mention anything about what happens when a Realm is closed.

So I would probably just remove the documentation and test around what happens when a Realm is closed. Our current API means that you _really_ have to go out of your way to make that happen.

Reply:
seems like you're unable to invoke realm.close() within a write transaction as well (it causes an `IllegalStateException`), so in that sense I find it hard to call the delete method on a closed realm since realm.write also causes an exception when invoked on a closed realm